File: python/nth-prime/nth_prime.py
def nth_prime(n):
	if n < 1: raise ValueError
	candidate = 2
	primes = []
	while len(primes) < n:
		if is_prime(candidate):
			primes.append(candidate)
		candidate += 1
	return primes.pop()


def is_prime(n):
	for i in [2, 3, 5, 7, 9]:
		if (n % i == 0):
			return n == i
	return all(False for i in range(11, n, 2) if (n % i == 0))

File: javascript/nth-prime/nth-prime.js
const LOW_PRIMES = [2, 3, 5, 7, 9, 13];

const range = (end, start = 0) => [...Array(end + 1).keys()].slice(start);

const isPrime = (n) => {
    if (n < 7) {
        return LOW_PRIMES.includes(n);
    }
    if (n % 2 === 0) {
        return false;
    }
    const max = Math.floor(Math.sqrt(n));
    return !range(max, 2)
        .filter(x => x % 2)
        .some(i => (n % i === 0));
};

module.exports = {
    nth: (n) => {
        if (n < 1) {
            throw new Error('Prime is not possible');
        }
        let candidate = 2;
        let primes = [];
        while (primes.length < n) {
            if (isPrime(candidate)) {
                primes.push(candidate);
            }
            candidate++;
        }
        return primes.pop();
    }
};

